### PR TITLE
Update to ethers v5 in Xstate Wallet

### PR DESCRIPTION
**Pull request by lalexgap**
Update ethers to v5 in xstate wallet

### CODE REVIEW

File: packages/xstate-wallet/src/utils/contract-utils.ts [hunk 1]
```diff
@@ -20,9 +19,9 @@ export function tokenAddress(assetHolderAddress: string): string | undefined {
   throw 'TokenAddress not found';
 }
 
-export function getProvider(): Web3Provider | JsonRpcProvider {
+export function getProvider(): providers.Web3Provider | providers.JsonRpcProvider {
```
Comment (snario) [inline packages/xstate-wallet/src/utils/contract-utils.ts:22]: Can we maintain directly importing these symbols?

File: packages/xstate-wallet/src/chain.ts [hunk 2]
```diff
@@ -325,18 +320,18 @@ export class ChainWatcher implements Chain {
       to: NITRO_ADJUDICATOR_ADDRESS
     };
 
-    const response = await this.signer.sendTransaction(transactionRequest);
+    const response = await this.signer.sendTransaction(transactionRequest as any); // TODO: Seems to be an ethers typing issue around the nonce;
```
Comment (snario) [inline packages/xstate-wallet/src/chain.ts:323]: What's the issue exactly?

Comment (lalexgap) [reply packages/xstate-wallet/src/chain.ts:323]: https://github.com/ethers-io/ethers.js/issues/844

Comment (snario) [reply packages/xstate-wallet/src/chain.ts:323]: This may be due to different implementations of `BigNumber` in the repo

File: packages/xstate-wallet/src/chain.ts [hunk 3]
```diff
@@ -151,7 +152,7 @@ export class FakeChain implements Chain {
   }
 
   public depositSync(channelId: string, expectedHeld: string, amount: string) {
-    const current = (this.channelStatus[channelId] || {}).amount || bigNumberify(0);
+    const current = (this.channelStatus[channelId] || {}).amount || BigNumber.from(0);
```
Comment (snario) [inline packages/xstate-wallet/src/chain.ts:155]: In general I like to use `Zero` instead of `BigNumber.from(0)`. We should probably stick to one convention.

File: package.json [hunk 4]
```diff
@@ -60,7 +60,12 @@
...
   },
-  "workspaces": [
-    "packages/*"
-  ]
+  "workspaces": {
+    "packages": [
+      "packages/*"
+    ],
+    "nohoist": [
+      "**/ethers"
```
Comment (kerzhner) [inline package.json:68]: Is this because a few other packages are not yet upgraded to ethers v5?

File: packages/xstate-wallet/package.json [hunk 5]
```diff
@@ -30,7 +30,7 @@
     "@xstate/react": "1.0.0-rc.3",
     "async-lock": "1.2.2",
     "dexie": "3.0.0",
-    "ethers": "4.0.45",
+    "ethers": "5.0.0-beta.187",
```
Comment (kerzhner) [inline packages/xstate-wallet/package.json:33]: we should probably use the same version between the wallet and the simple-hub to avoid having to download 2 versions of the package. The hub currently uses 5.0.0-beta.184

File: packages/xstate-wallet/src/chain.ts [hunk 6]
```diff
@@ -324,10 +320,13 @@ export class ChainWatcher implements Chain {
       ...Transactions.createConcludePushOutcomeAndTransferAllTransaction(
         finalizationProof.flatMap(toNitroSignedState)
       ),
-      to: NITRO_ADJUDICATOR_ADDRESS
+      to: NITRO_ADJUDICATOR_ADDRESS,
+      nonce: undefined
```
Comment (kerzhner) [inline packages/xstate-wallet/src/chain.ts:324]: Curious, why is this needed?

File: packages/xstate-wallet/src/chain.ts [hunk 7]
```diff
@@ -444,3 +447,21 @@ export class ChainWatcher implements Chain {
     );
   }
 }
+
+// Since nitro-protocol is still using v4 of ethers we need to convert any bignumbers the v5 version from ethers
+// TODO: Remove this when nitro protocol is using v5 ethers
```
Comment (kerzhner) [inline packages/xstate-wallet/src/chain.ts:452]: We might want to create an issue

File: packages/xstate-wallet/src/utils/contract-utils.ts [hunk 8]
```diff
@@ -24,22 +25,16 @@ export function getProvider(): Web3Provider | JsonRpcProvider {
...
-    return new providers.JsonRpcProvider(`http://localhost:${process.env.GANACHE_PORT}`);
+    return new JsonRpcProvider(`http://localhost:${process.env.GANACHE_PORT}`);
   }
 }
 
 export function getEthAssetHolderContract() {
   const provider = getProvider();
   return new Contract(
+    // eslint-disable-next-line no-process-env
     process.env.ETH_ASSET_HOLDER_ADDRESS || '0x0',
```
Comment (kerzhner) [inline packages/xstate-wallet/src/utils/contract-utils.ts:36]: Is there a reason why a constant from the config cannot be used here?

File: packages/xstate-wallet/src/workflows/approve-budget-and-fund.ts [hunk 9]
```diff
@@ -219,7 +219,7 @@ const createLedger = (store: Store) => async (context: Initial): Promise<LedgerI
 
   const stateVars = {
     outcome: initialOutcome,
-    turnNum: bigNumberify(0),
+    turnNum: BigNumber.from(0),
```
Comment (kerzhner) [inline packages/xstate-wallet/src/workflows/approve-budget-and-fund.ts:222]: Can we use Zero from '@ethersproject/constants' for code like this? (basically +1 to @snario's comment)

File: packages/xstate-wallet/src/utils/contract-utils.ts (hunk 8 above)
Comment (snario) [reply packages/xstate-wallet/src/utils/contract-utils.ts:36]: To be honest I’m surprised this is even passing linting. I thought I disabled using process.env outside of the config file.

Comment (snario) [reply packages/xstate-wallet/src/utils/contract-utils.ts:36]: Oh I didn’t even see the disable comment. Yeah let’s not do that :p

File: package.json (hunk 4 above)
Comment (lalexgap) [reply package.json:68]: Yup. That meant both versions of `ethers` were available which caused me to run into this [issue](https://github.com/ethers-io/ethers.js/issues/844).

It also helped me catch references to v4 of `ethers` that we hadn't removed from the `xstate-wallet` and `simple-hub`.

File: packages/xstate-wallet/src/chain.ts (hunk 6 above)
Comment (lalexgap) [reply packages/xstate-wallet/src/chain.ts:324]: Oops, I think I that was from debugging that I forgot to remove.

File: packages/xstate-wallet/src/utils/contract-utils.ts (hunk 8 above)
Comment (lalexgap) [reply packages/xstate-wallet/src/utils/contract-utils.ts:36]: I'm not sure how it survived linting prior to this PR 😕

File: packages/xstate-wallet/src/workflows/approve-budget-and-fund.ts (hunk 9 above)
Comment (lalexgap) [reply packages/xstate-wallet/src/workflows/approve-budget-and-fund.ts]: [21 files later](https://github.com/statechannels/monorepo/pull/1934/commits/0a24f385a509b09cb070beb23ea1e58bd2fb3f11) 😛
